client/src/backend/remote_node.rs

```diff
@@ -197,6 +197,18 @@ impl backend::Backend for RemoteNode {
         Ok(keys.into_iter().map(|key| key.0).collect())
     }
 
+    async fn block_header(&self, block_hash: Option<BlockHash>) -> Result<Header, Error> {
```

Comment:
The problem of using type alias is that it's never enforced, creating inconsistencies that leads to cases like this.
This to say:

```suggestion
    async fn block_header(&self, block_hash: Option<BlockHash>) -> Result<BlockHeader, Error> {
```

Reply:
Fixed